app/src/main/java/com/kamron/pogoiv/Pokefly.java

```diff
@@ -485,6 +491,8 @@ private void hideInfoLayoutArcPointer() {
 
     @Override
     public void onDestroy() {
+        LocalBroadcastManager.getInstance(this).sendBroadcast(new Intent(ACTION_STOP));
+
```

Comment:
Hmm... do you need isRunning() if you have to use Intents? I haven't fully gone through the code, but just pointing out for now.

If the receiver of the intent checks for Pokemon.isRunning() they'll get true. Is that the intended design? I hope this intent doesn't get MainActivity to try and stop this service but instead just updates the notification or something. Otherwise, it could try to stop the Pokefly service while it's already OnDestroying -- not sure if that's an issue in Android or not.